src/api/api.service.ts

```diff
@@ -67,15 +69,28 @@ export class ApiService {
           ),
         );
       }
-      return api;
+      if (api && isOwner === true) {
+        return api;
+      } else {
+        const {
+          ['base_url']: unused1,
+          ['visibility']: unused2,
+          ['subscriptions']: unused3,
+          ['status']: unused4,
+          ['secretKey']: unused5,
+          ...partialApi
+        } = api;
```

Comment:
Frankly speaking, I don't know what you are doing

Reply:
removing data not needed if not profileId